x-pack/plugins/maps/server/data_indexing/get_indexes_matching_pattern.ts

```diff
@@ -20,14 +21,20 @@ export async function getMatchingIndexes(
     const matchingIndexes = indexResults
       .map((indexRecord) => indexRecord.index)
       .filter((indexName) => !!indexName);
-    return {
-      success: true,
-      matchingIndexes: matchingIndexes as string[],
-    };
+    return response.ok({ body: { success: true, matchingIndexes: matchingIndexes as string[] } });
   } catch (error) {
-    return {
-      success: false,
-      error,
-    };
+    const errorStatusCode = error.meta?.statusCode;
+    if (errorStatusCode === 404) {
```

Comment:
Should 404 error be masked? That way the problem could still be debugged on the client. I think its best to just keep the original behavior and any error return the old payload of `{ success: false, error}`.

Reply:
It's not really a problem to be debugged as far as this endpoint is concerned. We get this when elasticsearch is up and running but returns no indexes match the criteria. So for ES it's an error, for this endpoint, it's an expected result